common/src/main/java/com/wynntils/services/secrets/SecretsService.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © Wynntils 2025.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.secrets;
+
+import com.google.gson.reflect.TypeToken;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Service;
+import com.wynntils.services.secrets.type.WynntilsSecret;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.io.FileUtils;
+
+public class SecretsService extends Service {
+    private static final File SECRETS_DIR = WynntilsMod.getModStorageDir("secrets");
+    private static final String SECRETS_FILE_NAME = "secrets.json";
+    private static final String MASTER_KEY_FILE_NAME = "master_key.json";
```

Comment:
```suggestion
    private static final String MASTER_KEY_FILE_NAME = "master_key.dat";
```

since you store raw base64 data and not json

Reply:
Or should we not even give it any file extension at all?